Mathlib/Topology/Order/Category/FrameAdjunction.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2023 Anne Baanen, Sam v. Gool, Leo Mayer, Brendan S. Murphy. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Sam v. Gool, Leo Mayer, Brendan S. Murphy
+-/
+import Mathlib.Topology.Category.Locale
+
+/-!
+# Adjunction between Locales and Topological Spaces
+
+This file defines functors between the categories of Locales and Topological Spaces
+and proves that they form an adjunction.
+
+## Main declarations
+
+* `pt`: the *points* functor from the category of locales to the category of topological spaces.
+* `topToLocale`: the forgetful functor from the category of topological spaces to the category of
+  locales.
+
+- `locale_top_adjunction`: the theorem that topToLocale is left adjoint to pt.
+
+## Motivation
+
+This adjunction provides a framework in which several Stone-type dualities fit.
+
+## Implementation notes
+
+- In naming the various functions below, we follow common terminology and reserve the word *point*
+  for an inhabitant of a type `X` which is a topological space, while we use the word *element* for
+  an inhabitant of a type `L` which is a locale.
+
+
+## References
+
+* [J. Picado and A. Pultr, Frames and Locales: topology without points][picado2011frames]
+
+## Tags
+
+topological space, frame, locale, Stone duality, adjunction, points
+-/
+
+open CategoryTheory Order Set Topology TopologicalSpace
+
+namespace Locale
+
+/- ### Definition of the points functor `pt` --/
+
+section pt_definition
+
+variable (L : Type*) [CompleteLattice L]
+
+/-- The type of points of a complete lattice `L`, where a *point* of a complete lattice is,
+by definition, a frame homomorphism from `L` to `Prop`. -/
+@[reducible]
+def PT := FrameHom L Prop
+
+/-- The frame homomorphism from a complete lattice `L` to the complete lattice of sets of
+points of `L`. -/
+@[simps]
+def openOfElementHom : FrameHom L (Set (PT L)) where
+  toFun u := {x | x u}
+  map_inf' a b := by simp [Set.setOf_and]
+  map_top' := by simp
+  map_sSup' S := by ext; simp [Prop.exists_iff]
+
+namespace PT
+
+/-- The topology on the set of points of the complete lattice `L`. -/
+instance instTopologicalSpace : TopologicalSpace (PT L) where
+  IsOpen s := ∃ u, {x | x u} = s
+  isOpen_univ := ⟨⊤, by simp⟩
+  isOpen_inter := by rintro s t ⟨u, rfl⟩ ⟨v, rfl⟩; use u ⊓ v; simp_rw [map_inf]; rfl
+  isOpen_sUnion S hS := by
+    choose f hf using hS
+    use ⨆ t, ⨆ ht, f t ht
+    simp_rw [map_iSup, iSup_Prop_eq, setOf_exists, hf, sUnion_eq_biUnion]
+
+/-- Characterizes when a subset of the space of points is open. -/
+lemma isOpen_iff (U : Set (PT L)) : IsOpen U ↔ ∃ u : L, {x | x u} = U := Iff.rfl
+
+end PT
+
+/-- The contravariant functor `pt` from the category of locales to the category of
```

Comment:
Isn't it covariant?

Reply:
Oops yes. At some point the category of frames was changed into the category of locales (not by me) but the doc strings weren't updated accordingly...